platform/inputs/analyze.go

```diff
@@ -0,0 +1,144 @@
+package inputs
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle"
+	"github.com/buildpacks/lifecycle/api"
+	"github.com/buildpacks/lifecycle/buildpack"
+	"github.com/buildpacks/lifecycle/image"
+	"github.com/buildpacks/lifecycle/internal/str"
+)
+
+// Analyze holds the values of command-line flags and args.
+// Fields are the cumulative total of inputs across all supported platform APIs.
+type Analyze struct {
```

Comment:
For simplicity and ease of comprehension, maybe this file and analyzer_factory.go should be one file. The idea is basically to create a pipeline from user-provided flags and args to a `lifecycle.Analyzer`.